plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,230 @@
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnMetadata;
+import static io.trino.plugin.hive.HiveMetadata.TABLE_COMMENT;
+import static io.trino.plugin.hive.metastore.MetastoreUtil.buildInitialPrivilegeSet;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.function.BinaryOperator.maxBy;
+
+public class DeltaLakeTableMetadataScheduler
+{
+    private static final Logger log = Logger.get(DeltaLakeTableMetadataScheduler.class);
+
+    private static final String TRINO_LAST_TRANSACTION_VERSION = "trino_last_transaction_version";
+    private static final String TRINO_METADATA_SCHEMA_STRING = "trino_metadata_schema_string";
+
+    private final HiveMetastoreFactory hiveMetastoreFactory;
+    private final TypeManager typeManager;
+    private final int tableParameterLengthLimit;
+    private final OptionalInt storeTableMetadataThreads;
+    private final Duration storeTableMetadataInterval;
+    private final Map<SchemaTableName, UpdateInfo> updateInfos = new ConcurrentHashMap<>();
+    private final boolean enabled;
+
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduler;
+
+    @Inject
+    public DeltaLakeTableMetadataScheduler(
+            HiveMetastoreFactory hiveMetastoreFactory,
+            NodeManager nodeManager,
+            TypeManager typeManager,
+            @TableParameterLengthLimit int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.hiveMetastoreFactory = requireNonNull(hiveMetastoreFactory, "hiveMetastoreFactory is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        this.storeTableMetadataInterval = config.getStoreTableMetadataInterval();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    public void putAll(Map<SchemaTableName, UpdateInfo> tableParameters)
+    {
+        updateInfos.putAll(tableParameters);
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            int threads = storeTableMetadataThreads.orElse(0);
+            executor = threads == 0 ? newDirectExecutorService() : newFixedThreadPool(threads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+            scheduler.scheduleWithFixedDelay(this::process, 0, storeTableMetadataInterval.toMillis(), MILLISECONDS);
```

Comment:
this needs to catch and log errors. Any uncaught exception will probably not be logged at all, and will also prevent the scheduler from repeating the process calls.

follow this pattern
https://github.com/trinodb/trino/blob/f37d10d23193255241de1ebacea83fcd6d42c3c5/core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java#L282-L295